lib/importer/index.js

```diff
@@ -2,141 +2,221 @@ const multer = require("multer");
 const session_lib = require("./session.js");
 const sheets_lib = require("./sheets.js");
 
-exports.Initialise = (config, router) => {
-  const storage = multer.diskStorage({
-    destination: (req, file, cb) => {
-      var fs = require("fs");
-      if (!fs.existsSync(config.uploadPath)) {
-        fs.mkdirSync(config.uploadPath, { recursive: true });
-      }
+const IMPORTER_SESSION_KEY = "importer.session"
+const IMPORTER_ERROR_KEY = "importer.error"
+
+//--------------------------------------------------------------------
+// The endpoints used to receive POST requests from importer macros, 
+// and used to generate template functions that can be used to find 
+// the URL for each step (to be used in form actions)
+//--------------------------------------------------------------------
+const IMPORTER_ROUTE_MAP = new Map([
+  ["importerStartPath", "/importer/start"],
```

Comment:
I'm nervous that "importer" might be too general a name to inject into app-global namespaces. People never have to type these by hand, so would something like "spreadsheet-importer" be fine? Or am I worrying too much?

Reply:
I think I agree, I just can't think of a better name to use and we're annoyingly constrained in the patterns we can use for namespacing (e.g no -). I'll see if I can get concensus on naming and then submit a PR with names changed.